### DISCUSSION TITLE
Using dataset_scenario generators

### DISCUSSION THREAD

**Discussion by Nidarshan619** [ContinualAI/avalanche#475] [Q&A] [answered]
I am finding it difficult to use my dataset in scenario generation. Please can you advise me

![image](https://user-images.githubusercontent.com/55737586/113146401-e9e2b880-922f-11eb-9ba8-c0de6bc88feb.png)

In the figure, you see 4 datasets train_MNIST, train_cifar10, test _dataset, and train_dataset. I am trying to bring train and test datasets into train_MNIST and train_cifar10 form. So that it can be added into the below code for scenario generation. 

 `generic_scenario =  dataset_scenario(
    train_dataset_list=[train_dataset],
    test_dataset_list=[test_dataset],
    task_labels=[0, 1]
)
`
Trying the above code on my dataset. I got the following error.

`ValueError: Unsupported dataset: must have a valid targets fieldor has to be a Tensor Dataset with at least 2Tensors`

Please can you assist me with some examples

**Comment by lrzpellegrini** [comment, marked as answer]
Hi there, the datasets you pass to `dataset_scenario` must be datasets containing a valid `targets` field (containing the class label for each pattern) or instances of `Subset` and `ConcatDataset` pointing to a dataset with that field.

First, are `train_dataset` and `test_dataset` subsets of MNIST and Cifar datasets? Or are they subsets of some other dataset?

**Reply by vlomonaco** [reply]
@Nidarshan619 Do you need further help on this?

**Reply by vlomonaco** [reply]
Solving this for inactivity! @Nidarshan619 let us know if you want to reopen it.
